lib/src/component_declaration/component_base.dart

```diff
@@ -451,7 +445,7 @@ abstract class UiProps extends MapBase
   /// Is a noop if [shouldAdd] is `false` or [propMap] is `null`.
   ///
   /// > Related: [addProp], [modifyProps]
-  void addProps(Map propMap, [bool shouldAdd = true]) {
+  void addProps(Map? propMap, [bool shouldAdd = true]) {
```

Comment:
We've always allowed the map to be nullable for convenience.

For example, it's handy when you have an optional props map prop, and want to do something like
```dart
return (Bar()
  ..addProps(props.barProps)
)();
```
While we could explore making this non-nullable, I don't think it'd be worthwhile due to how much it'd impact existing consumers, especially since they won't be migrated to null safety and the analyzer won't be able to tell them they can't pass nullable values in.